.github/workflows/client.yml

```diff
@@ -49,76 +49,37 @@ jobs:
     runs-on: ubuntu-18.04
     needs: build
     if: github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        environment:
+          [
+            { "name": "ambroise", "secret": "IP_VM_AMB" },
+            { "name": "aphp", "secret": "IP_VM_AP" },
+            { "name": "clb", "secret": "IP_VM_CLB" },
+            { "name": "curie", "secret": "IP_VM_CURIE" },
+            { "name": "demo", "secret": "" },
+            { "name": "dev", "secret": "" },
+            { "name": "rms", "secret": "IP_VM_REIMS" },
+            { "name": "staging", "secret": "" },
+          ]
     steps:
       - uses: actions/checkout@v1
 
-      - name: Build staging and demo
-        run: |
-          cd client/
-          yarn build:docker:staging
-          yarn build:docker:dev
-          yarn build:docker:demo
-
-      - name: Build Ambroise
-        env:
-          IP: ${{ secrets.IP_VM_AMB }}
-          STAGE: ambroise
-        run: |
-          cd client
-          ./scripts/generate_client_env.sh > .env.ambroise
-          docker build -t arkhn/pyrog-client:ambroise --build-arg ENV_FILE=.env.ambroise .
-
-      - name: Build AP
-        env:
-          IP: ${{ secrets.IP_VM_AP }}
-          STAGE: aphp
-        run: |
-          cd client
-          ./scripts/generate_client_env.sh > .env.aphp
-          docker build -t arkhn/pyrog-client:aphp --build-arg ENV_FILE=.env.aphp .
-
-      - name: Build Clb
-        env:
-          IP: ${{ secrets.IP_VM_CLB }}
-          STAGE: clb
-        run: |
-          cd client
-          ./scripts/generate_client_env.sh > .env.clb
-          docker build -t arkhn/pyrog-client:clb --build-arg ENV_FILE=.env.clb .
-
-      - name: Build Curie
-        env:
-          IP: ${{ secrets.IP_VM_CURIE }}
-          STAGE: curie
-        run: |
-          cd client
-          ./scripts/generate_client_env.sh > .env.curie
-          docker build -t arkhn/pyrog-client:curie --build-arg ENV_FILE=.env.curie .
-
-      - name: Build RMS
+      - name: Build image
         env:
-          IP: ${{ secrets.IP_VM_REIMS }}
-          STAGE: rms
+          IP: ${{ secrets[matrix.environment.secret] }}
```

Comment:
Isn't this "" for demo, staging and dev?

Reply:
yup, this wont work as is